cognite_toolkit/_cdf_tk/storageio/_asset_centric.py

```diff
@@ -169,8 +169,10 @@ class AssetIO(BaseAssetCentricIO[str, AssetWrite, Asset, AssetWriteList, AssetLi
     UPLOAD_ENDPOINT = "/assets"
 
     def as_id(self, item: dict[str, JsonVal] | object) -> int:
-        if isinstance(item, Asset | AssetWrite) and item.id is not None:  # type: ignore[union-attr]
-            return item.id  # type: ignore[union-attr]
+        if isinstance(item, Asset) and item.id is not None:
+            return item.id
+        elif isinstance(item, AssetWrite):
+            return hash(item.external_id)
```

Comment:
This is very very ugly, but will be replaced in the next PR and is currently not in use. It is the simplest way I can think of to avoid bloating this PR even more.